lld/ELF/Relocations.cpp

```diff
@@ -1357,8 +1357,8 @@ static unsigned handleTlsRelocation(RelType type, Symbol &sym,
             R_LOONGARCH_GOT_PAGE_PC, R_GOT_OFF, R_TLSIE_HINT>(expr)) {
     ctx.hasTlsIe.store(true, std::memory_order_relaxed);
     // Initial-Exec relocs can be relaxed to Local-Exec if the symbol is locally
-    // defined.
-    if (toExecRelax && isLocalInExecutable) {
+    // defined.  This is not supported on SystemZ.
+    if (toExecRelax && isLocalInExecutable && config->emachine != EM_S390) {
```

Comment:
Update `bool toExecRelax` declaration instead?

Reply:
But that would disable *all* the relaxations.  We do have Local-Dynamic to Local-Exec and Gobal-Dynamic to Initial/Local-Exec.  We just cannot do Initial-Exec to Local-Exec relaxation.